src/mds/MDCache.cc

```diff
@@ -12933,3 +12936,23 @@ bool MDCache::dump_inode(Formatter *f, uint64_t number) {
   f->close_section();
   return true;
 }
+
+void MDCache::handle_mdsmap(const MDSMap &mdsmap) {
+  // process export_pin_delayed_queue whenever a new MDSMap received
+  auto &q = export_pin_delayed_queue;
+  auto it = q.begin();
+  while (it != q.end()) {
```

Comment:
nit:

`for (auto it = q.begin(); it != q.end(); ) {`

would be preferable to limit the scope of `it`.